lesson 6/Tensor/app/src/main/cpp/native-lib.cpp

```diff
@@ -0,0 +1,95 @@
+#include <jni.h>
+#include <string>
+#include <string.h>
+#include <sys/socket.h>	//socket
+#include <arpa/inet.h>	//inet_addr
+#include <unistd.h>
+#include <android/log.h>
+#include <errno.h>
+#include <locale>
+
+
+#define LOG_TAG "CLIENT_APP"
+
+jbyteArray getImage(jobject context, JNIEnv* env) {
+    jclass contextThemeWrapperClass = env->FindClass("android/view/ContextThemeWrapper");
+    jclass assetManagerClass = env->FindClass("android/content/res/AssetManager");
+    jclass inputStreamClass = env->FindClass("java/io/InputStream");
+
+    jmethodID getAssetsID = env->GetMethodID(contextThemeWrapperClass, "getAssets", "()Landroid/content/res/AssetManager;");
+    jmethodID openID = env->GetMethodID(assetManagerClass, "open", "(Ljava/lang/String;)Ljava/io/InputStream;");
+    jmethodID availableID = env->GetMethodID(inputStreamClass, "available", "()I");
+    jmethodID readID = env->GetMethodID(inputStreamClass, "read", "([B)I");
+    jmethodID closeID = env->GetMethodID(inputStreamClass, "close", "()V");
+
+    jobject assets = env->CallObjectMethod(context, getAssetsID);
+    jobject stream = env->CallObjectMethod(assets, openID, env->NewStringUTF("zivert.jpg"));
```

Comment:
Your implementation of this line was like this:
`jobject open = env->CallObjectMethod(context, openFile);`
There are a few issues here:
1. This call returns the `stream` object (you can look at the corresponding java code line), so you should name `jobject stream`, not `jobject open`.
2. This function (open function) is called on the `assets` object not on `context`.
3. It takes a string argument (you can look at the corresponding java line), so you should pass a string parameter with `NewStringUTF` method